tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -6896,12 +6910,26 @@ uint32_t SPIRVEmitter::translateAPInt(const llvm::APInt &intValue,
     }
   }
   case 32: {
-    if (isSigned)
+    if (isSigned) {
+      if (!intValue.isSignedIntN(32)) {
+        emitError("evaluating this integer literal as a 32-bit integer loses "
```

Comment:
We are not outputting the source code location here. "this integer literal" points to nowhere. Should output the contents in the literal number: `evaluating integer literal '%0' as ..`